### PR TITLE
 Move include/friendica_smarty to src/

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #3878

This task involved fixing a potential infinite loop when the template engine registration fails, calling `killme()` which spawns a new App since the current one isn't done spawning yet, during which the template engine registration fails, calling `killme()`, which spawns a new App since the current one isn't done spawning yet, etc...

### CODE REVIEW

**Comment by zeroadam** [comment]
This one was on my tried and failed list!!

File: src/Render/FriendicaSmarty.php [hunk 1]
```diff
@@ -1,96 +1,54 @@
...
+	{
+		parent::__construct();
+
+		$a = get_app();
+		$theme = current_theme();
+
+		// setTemplateDir can be set to an array, which Smarty will parse in order.
+		// The order is thus very important here
+		$template_dirs = ['theme' => "view/theme/$theme/" . SMARTY3_TEMPLATE_FOLDER . "/"];
+		if (x($a->theme_info, "extends"))
```
Comment (tobiasd) [inline src/Render/FriendicaSmarty.php]: shouldn't here be curly brackets to encapsulate the if contents?

**Comment by tobiasd** [comment]
After pulling this in I get

    replace_macros: Unable to load template 'file:view/SMARTY3_TEMPLATE_FOLDER/group_side.tpl'

any ideas?

**Comment by tobiasd** [comment]
Adding

    define('SMARTY3_TEMPLATE_FOLDER', 'templates');

to the FriendicaSmartyEngine file fixes this problem. But is it the correct way? I think a constant should only be defined once.

**Comment by zeroadam** [comment]
Same here. I feel like this merge was at least after the overnight pulls done in "The East" so hopefully we don't have dozens of wsod nodes at the moment!

**Pull request merged**